### PR TITLE
validate auth token for all api endpoints by default

### PR DESCRIPTION
**Pull request by arbll**
### What does this PR do?

Validates the auth token for all api endpoints by default using a middleware.

### Motivation

Adding validation for every endpoint was error-prone.

### CODE REVIEW

**Comment by trishankatdatadog** [comment]
Thoughts (thanks also to @pratikgs-dd for brainstorming):

* Should server regenerate the token on every restart?
* Should clients authenticate server (using the pubkey of the server self-signed TLS cert)?

**Comment by trishankatdatadog** [comment]
Here are some good recommendations we could use from the OAuth 2.0 RFC for Bearer Tokens: https://tools.ietf.org/html/rfc6750#section-5.3

File: cmd/agent/api/agent/agent_jmx.go [hunk 1]
```diff
@@ -57,22 +50,22 @@ func getJMXConfigs(w http.ResponseWriter, r *http.Request) {
 	for name, config := range configItems {
 		m, ok := config.(map[string]interface{})
 		if !ok {
-			err = fmt.Errorf("wrong type in cache")
+			err := fmt.Errorf("wrong type in cache")
```
Comment (trishankatdatadog) [inline cmd/agent/api/agent/agent_jmx.go]: Not a Go expert, but how did this code used to work before?

Comment (truthbk) [reply cmd/agent/api/agent/agent_jmx.go]: different scope

**Review by truthbk — DISMISSED** [review]
Super elegant approach, big fan of how you used the middlewares feature in gorillamux! 🍰 💯

File: cmd/agent/api/server.go [hunk 2]
```diff
@@ -100,3 +103,12 @@ func StopServer() {
 func ServerAddress() *net.TCPAddr {
 	return listener.Addr().(*net.TCPAddr)
 }
+
+func validateToken(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if err := util.Validate(w, r); err != nil {
+			return
```
[the remark this answers is not in the document]
Comment (arbll) [reply cmd/agent/api/server.go]: [`Validate` handles that](https://github.com/DataDog/datadog-agent/blob/0aa3114a92a568b8b84a02919789fbc2d9358477/pkg/api/util/util.go#L61-L85)

File: cmd/agent/api/agent/agent_jmx.go (hunk 1 above)
Comment (arbll) [reply cmd/agent/api/agent/agent_jmx.go]: before this diff `err` was declared above and reused in the following code blocks

**Comment by arbll** [comment]
Hey @trishankatdatadog 
While I agree with most of your thoughts I think we should keep this PR in its scope.
Let's move them to the issue repo and discuss them there. Is that good with you?

**Comment by trishankatdatadog** [comment]
@arbll sure thing

**Review by truthbk — APPROVED** [review]
Go for it! 🚀

**Pull request merged**